packages/flutter_tools/templates/list_detail_app/lib/src/settings/settings_controller.dart.tmpl

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:{{projectName}}/src/settings/settings_service.dart';
+
+/// A class that many Widgets can interact with to read user settings, update
+/// user settings, or listen to user settings changes.
+///
+/// Controllers glue Data Services to Flutter Widgets. The SettingsController
+/// uses the SettingsService to store and retrieve user settings.
+class SettingsController with ChangeNotifier {
```

Comment:
why `with` vs `extends`?

Reply:
I actually don't have a strong preference here. 

I went with a mixin since it's a bit more "flexible" in that end-users could extend from another class of their choice if need be. 

The [documentation of the class](https://api.flutter.dev/flutter/foundation/ChangeNotifier-class.html) simply reads "A class that can be extended or mixed in," but is extending the recommended approach?

Reply:
`with` is fine, I was just curious.